projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -74,6 +74,20 @@ public function get_preferred_view( $screen, $fallback_global_preference = true
 		return parent::get_preferred_view( $screen, $fallback_global_preference );
 	}
 
+	/**
+	 * Point the Site Editor's `< Dashboard` link to wpcom home.
+	 *
+	 * Although this isn't strictly an admin menu item, it belongs here because it's part of
+	 * changing wp-admin links to their wp.com equivalents.
+	 *
+	 * @param  array $settings Editor settings.
+	 * @return array           Updated Editor settings.
+	 */
+	public function site_editor_dashboard_link( $settings ) {
+		$settings['__experimentalDashboardLink'] = 'https://wordpress.com/home/' . $this->domain;
```

Comment:
It looks like this may be causing this issue https://github.com/Automattic/wp-calypso/issues/71760 . 

In short in testing flows on horizon, leaving the site editor on horizon now bumps you into the production environment.